heed/src/env.rs

```diff
@@ -830,6 +830,74 @@ mod tests {
             .unwrap();
     }
 
+    #[test]
+    fn open_env_with_named_path_hardlink_and_no_subdir() {
+        let dir = tempfile::tempdir().unwrap();
+        let env_name = dir.path().join("babar.mdb");
+        let hardlink_name = dir.path().join("babar.mdb.link");
+
+        let mut envbuilder = EnvOpenOptions::new();
+        unsafe { envbuilder.flag(crate::Flag::NoSubDir) };
+        let _env = envbuilder
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&env_name)
+            .unwrap();
+
+        std::fs::hard_link(&env_name, &hardlink_name).unwrap();
+        let _env = envbuilder
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&hardlink_name)
+            .unwrap();
+
+        let _env = envbuilder
+            .map_size(10 * 1024 * 1024) // 10MB
+            .open(&env_name)
+            .unwrap();
+    }
+
```

Comment:
This test is flawed, since we follow only `directory` and hardlink only apply to env it's just not-appliable.

Reply:
Can we have a symlink test with directories to replace this one? Obviously only running on UNIX.

Reply:
Yeah already have one. :)
https://github.com/meilisearch/heed/pull/179/files#diff-ac577a0bca84e6fbce36b3a413107331b37c74809219009e740896600a1d97f7R861-R881

`symlink_dir` exist for windows two already did a test:
https://github.com/meilisearch/heed/pull/179/files#diff-ac577a0bca84e6fbce36b3a413107331b37c74809219009e740896600a1d97f7R982-R1002


May push the changes you suggested on test this morning.